File: sansa-rdf/sansa-rdf-common/src/main/scala/net/sansa_stack/rdf/common/kryo/jena/ByteArrayUtils.scala
package net.sansa_stack.rdf.common.kryo.jena

import java.nio.ByteBuffer

import com.esotericsoftware.kryo.io.{Input, Output}

object ByteArrayUtils {
  def write(output: Output, bytes: Array[Byte]): Unit = {
    output.writeInt(bytes.length, true)
    output.writeBytes(bytes)
  }

  def write(output: Output, bytes: Array[Byte], offset: Int, length: Int): Unit = {
    output.writeInt(length, true)
    output.writeBytes(bytes, offset, length)
  }

  def read(input: Input): Array[Byte] = {
    val len = input.readInt(true)
    val result = input.readBytes(len)
    result
  }
}

File: sansa-kryo-jena/src/main/java/net/sansa_stack/kryo/jena/ByteArrayUtils.java
package net.sansa_stack.kryo.jena;

import com.esotericsoftware.kryo.io.Input;
import com.esotericsoftware.kryo.io.Output;

/**
 * Utils for writing and reading byte arrays by prefixing the data with its length
 *
 * @author Claus Stadler
 */
public class ByteArrayUtils {
    public static void write(Output output, byte[] bytes) {
        output.writeInt(bytes.length, true);
        output.writeBytes(bytes);
    }

    public static void write(Output output, byte[] bytes, int offset, int length) {
        output.writeInt(length, true);
        output.writeBytes(bytes, offset, length);
    }

    public static byte[] read(Input input) {
        int len = input.readInt(true);
        byte[] result = input.readBytes(len);
        return result;
    }
}
